frontend/src/metabase/search/components/SidebarFilter/SidebarFilter.styled.tsx

```diff
@@ -1,37 +1,70 @@
 import styled from "@emotion/styled";
 import { css } from "@emotion/react";
-import { ParameterFieldSet } from "metabase/parameters/components/ParameterWidget/ParameterWidget.styled";
-
-export const DropdownFilterElement = styled(ParameterFieldSet)`
-  height: 40px;
-
-  ${({ theme, fieldHasValueOrFocus }) => {
-    return (
-      fieldHasValueOrFocus &&
-      css`
-        border-color: ${theme.colors.brand[1]};
-        color: ${theme.colors.brand[1]};
-      `
-    );
-  }}
-  &:hover {
-    ${({ theme }) => {
-      return css`
-        background-color: ${theme.colors?.bg[1]};
-        transition: background-color 0.3s;
-        cursor: pointer;
-      `;
-    }}
+import type { HTMLAttributes } from "react";
+import type { ButtonProps } from "metabase/ui";
+import { Button, Stack } from "metabase/ui";
+import FieldSet from "metabase/components/FieldSet";
+import EventSandbox from "metabase/components/EventSandbox";
+
+export const DropdownFieldSet = styled(FieldSet)<{
+  fieldHasValueOrFocus?: boolean;
+}>`
+  min-width: 0;
+  text-overflow: ellipsis;
+  overflow: hidden;
+
+  border: 2px solid
+    ${({ theme, fieldHasValueOrFocus }) =>
+      fieldHasValueOrFocus ? theme.colors.brand[1] : theme.colors.border[0]};
+
+  margin: 0;
+  padding: 0.5rem 0.75rem;
+
+  cursor: pointer;
+
+  legend {
+    min-width: 0;
+    max-width: 100%;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+
+    text-transform: none;
+    position: relative;
+    height: 2px;
+    line-height: 0;
+    margin-left: -0.45em;
+    padding: 0 0.5em;
```

Comment:
Part of this code is duplicated from `ParameterFieldSet` - we should probably keep it in one place (now or later). Notice that the overflow does not occur in `ParameterFieldSet` and is handled differently (text wraps) than here (ellipsis) - an inconsistency which would go away if there was no code duplication.


![image](https://github.com/metabase/metabase/assets/6830683/fd1b0212-ef16-4228-aca7-02b66afcd8e2)

Reply:
This is actually just an issue with the fact that I didn't truncate the text in `TypeFilterDisplay` - I'm not using `ParameterFieldSet` at all in this code anymore since there were overflow issues as you saw in the previous iteration. Probably a conversation about whether or not the `DisplayComponent` should exist but I'll keep it for now

Reply:
(All dropdown components are using the same code, just different components within the popover and within the fieldset display)